hs_file_types/models/csv.py

```diff
@@ -0,0 +1,757 @@
+                        html_tags.p(f"Description: {col_desc}")
+                        html_tags.p(f"Data type: {col.datatype}")
+                        html_tags.hr()
+
+            html_string += root_div.render()
+
+        template = Template(html_string)
+        context = Context({})
+        return template.render(context)
+
+    def get_html_forms(self, dataset_name_form=True, temporal_coverage=True, **kwargs):
+        root_div, context = super(CSVFileMetaData, self).get_html_forms(render=False)
+        with root_div:
+            with html_tags.div(cls="content-block"):
+                with html_tags.div():
+                    html_tags.legend("CSV File Properties:")
+                    table_schema_model = self.get_table_schema_model()
+                    form_action = f"/hsapi/_internal/{self.logical_file.id}/update-csv-table-schema/"
+                    with html_tags.form(id="id-csv-metadata", action=form_action, method="post",
+                                        encrtype="multipart/form-data"):
+                        column_input_cls = "form-control input-sm textinput textInput"
+                        row_div = html_tags.div(cls="control-group", id="id-csv-metadata-row-count")
+                        with row_div:
+                            html_tags.label("Number of data rows:", fr="id-csv-metadata-rows",
+                                            cls="control-label")
+                            with html_tags.div(cls="controls"):
+                                html_tags.input(type="text", id="id-csv-metadata-rows", name="rows",
+                                                readonly="readonly",
+                                                value=table_schema_model.rows,
+                                                cls=column_input_cls)
+                        delimiter_div = html_tags.div(cls="control-group", id="id-csv-metadata-row-count")
+                        with delimiter_div:
+                            delimiter = self._get_delimiter_name(table_schema_model.delimiter)
+                            html_tags.label("Delimiter:", fr="id-csv-metadata-delimiter",
+                                            cls="control-label")
+                            with html_tags.div(cls="controls"):
+                                html_tags.input(type="text", id="id-csv-metadata-delimiter", name="rows",
+                                                readonly="readonly",
+                                                value=delimiter,
+                                                cls=column_input_cls)
+                        col_div = html_tags.div(cls="control-group", id="id-csv-metadata-columns")
+                        with col_div:
+                            with html_tags.div():
+                                html_tags.legend("Column Properties:", style="font-size: 1.2em;")
+                                for col_no, col in enumerate(table_schema_model.table.columns):
```

Comment:
![image](https://github.com/user-attachments/assets/0372acb4-b72a-4249-b375-de36ec3f706a)
for future improvement, perhaps we could do a drop-down. Otherwise csv with large number of cols will be pretty unmanageable using the current form

Reply:
Agreed